### PR TITLE
cleanup: remove unused resources associated to former EC2 VM agents and their AMIs

### PR DESCRIPTION
**Pull request by dduportal**
Related to https://github.com/jenkins-infra/helpdesk/issues/3662, this PR removes resources unused for months since we moved away VM agents from AWS (ref. https://github.com/jenkins-infra/helpdesk/issues/3502).

### CODE REVIEW

File: ec2-agents.tf [hunk 1]
```diff
@@ -1,149 +0,0 @@
-## Identity for Jenkins Controller infra.ci.jenkins.io
-data "aws_iam_user" "jenkins_infra_ci" {
-  user_name = "jenkins-infra-ci"
-}
```
Comment (dduportal) [inline ec2-agents.tf:4]: Note: to be deleted manually (tracked in https://github.com/jenkins-infra/helpdesk/issues/3662#issuecomment-1770472278)

File: ec2-agents.tf [hunk 2]
```diff
@@ -1,149 +0,0 @@
...
-
-resource "aws_iam_user_policy_attachment" "allow_ec2_on_infraci" {
-  user       = data.aws_iam_user.jenkins_infra_ci.user_name
-  policy_arn = aws_iam_policy.jenkins_ec2_agents.arn
-}
-
-## Identity for Jenkins Controller ci.jenkins.io
-data "aws_iam_user" "jenkins_ci" {
-  user_name = "jenkins-ci"
-}
```
Comment (dduportal) [inline ec2-agents.tf:59]: Note: to be deleted manually (tracked in https://github.com/jenkins-infra/helpdesk/issues/3662#issuecomment-1770472278)

**Comment by dduportal** [comment]
```
Plan: 0 to add, 0 to change, 11 to destroy.
```

as expected

**Review by timja — APPROVED** [review]

**Review by lemeurherve — APPROVED** [review]

**Comment by dduportal** [comment]
For info, the terraform apply of this PR failed with the following error:

```
Error: removing policy arn:aws:iam::<redacted>:policy/jenkins_ec2_agents from IAM User jenkins-ci: AccessDenied: User: arn:aws:iam::<redacted>:user/<redacted> is not authorized to perform: iam:DetachUserPolicy on resource: user jenkins-ci because no identity-based policy allows the iam:DetachUserPolicy action
```

3 times (for the 3 users: `updatecli` , `jenkins-ci` and jenkins-infra-ci`).


Proposal: No need to add the permission `iam:DetachUserPolicy` to our technical users (in the private repo jenkins-infra/terraform-states) for safety reasons. Since this is a one-time operation, i'll detach the permissiosn from users and will re-trigger the terraform job

**Comment by dduportal** [comment]
Done with success!

**Pull request merged**